aiohttp/web_reqrep.py

```diff
@@ -381,6 +381,33 @@ def post(self):
         self._post = MultiDictProxy(out)
         return self._post
 
+    def clone(self, *, method=None, path=None, headers=None, raw_headers=None):
+        """
+        Creates and returns a new instance of Request object. If no parameters
+        are given, an exact copy is returned. If a parameter is not passed, it
+        will reuse the one from the current request object.
+
+        :param method: str http method
+        :param path: str url path to use
+        :param headers: CIMultidictProxy or compatible containing the headers.
+        :param raw_headers: tuple of two element tuples containing the headers
+            as bytearrays.
+        """
+
+        message = RawRequestMessage(
+            method or self.method, path or self.path, self.version,
+            headers or self.headers, raw_headers or self.raw_headers,
+            self.keep_alive, None)
```

Comment:
No. `headers` is `CIMultiDict` instance with `str` keys and values.
But `raw_headers` is a list or (header, value) pairs where both are unconverted `bytes`.
Not all possible header and value can be converted using UTF8 codec. Some old clients may send a data using custom encoding and the purpose of `raw_headers` is giving a way to handle this data.